bee-ledger/src/workers/storage.rs

```diff
@@ -0,0 +1,496 @@
+        + Batch<OutputId, ConsumedOutput>
+        + Batch<Unspent, ()>
+        + Batch<(), LedgerIndex>
+        + Batch<MilestoneIndex, OutputDiff>
+        + Batch<(Ed25519Address, OutputId), ()>
+        + Batch<Address, Balance>
+        + Batch<(MilestoneIndex, Receipt), ()>
+        + Batch<(bool, TreasuryOutput), ()>
+        + Exist<Unspent, ()>
+        + Fetch<(), SnapshotInfo>
+        + Fetch<OutputId, CreatedOutput>
+        + Fetch<(), LedgerIndex>
+        + Fetch<Address, Balance>
+        + Fetch<bool, Vec<TreasuryOutput>>
+        + Insert<(), SnapshotInfo>
+        + Insert<(), LedgerIndex>
+        + Insert<(bool, TreasuryOutput), ()>
+        + Truncate<SolidEntryPoint, MilestoneIndex>
+        + for<'a> AsStream<'a, Unspent, ()>
+        + for<'a> AsStream<'a, Address, Balance>
+        + for<'a> AsStream<'a, SolidEntryPoint, MilestoneIndex>
+        + bee_tangle::storage::StorageBackend
+{
+}
+
+pub(crate) fn insert_output_id_for_address_batch<B: StorageBackend>(
+    storage: &B,
+    batch: &mut <B as BatchBuilder>::Batch,
+    address: &Address,
+    output_id: &OutputId,
+) -> Result<(), Error> {
+    match address {
+        Address::Ed25519(address) => {
+            Batch::<(Ed25519Address, OutputId), ()>::batch_insert(storage, batch, &(*address, *output_id), &())
+                .map_err(|e| Error::Storage(Box::new(e)))?;
+        }
+        address => return Err(Error::UnsupportedAddressKind(address.kind())),
+    }
+
+    Ok(())
+}
+
+pub(crate) fn delete_output_id_for_address_batch<B: StorageBackend>(
+    storage: &B,
+    batch: &mut <B as BatchBuilder>::Batch,
+    address: &Address,
+    output_id: &OutputId,
+) -> Result<(), Error> {
+    match address {
+        Address::Ed25519(address) => {
+            Batch::<(Ed25519Address, OutputId), ()>::batch_delete(storage, batch, &(*address, *output_id))
+                .map_err(|e| Error::Storage(Box::new(e)))?;
+        }
+        address => return Err(Error::UnsupportedAddressKind(address.kind())),
+    }
+
+    Ok(())
+}
+
+pub(crate) fn insert_created_output_batch<B: StorageBackend>(
+    storage: &B,
+    batch: &mut <B as BatchBuilder>::Batch,
+    output_id: &OutputId,
+    output: &CreatedOutput,
+) -> Result<(), Error> {
+    Batch::<OutputId, CreatedOutput>::batch_insert(storage, batch, output_id, output)
+        .map_err(|e| Error::Storage(Box::new(e)))?;
+    Batch::<Unspent, ()>::batch_insert(storage, batch, &(*output_id).into(), &())
+        .map_err(|e| Error::Storage(Box::new(e)))?;
+
+    match output.inner() {
+        Output::SignatureLockedSingle(output) => {
+            insert_output_id_for_address_batch(storage, batch, output.address(), output_id)?
+        }
+        Output::SignatureLockedDustAllowance(output) => {
+            insert_output_id_for_address_batch(storage, batch, output.address(), output_id)?
+        }
+        output => return Err(Error::UnsupportedOutputKind(output.kind())),
+    }
+
+    Ok(())
```

Comment:
Remove the `?` from lines 133 and 136, the `return` from 138 and the `Ok(())` from 141.